src/code/sys_matrix.c

```diff
@@ -1,89 +1,1939 @@
+ *
+ * @param[out] dest output vector.
+ *
+ * @remark original name: "Matrix_Position_Zero"
+ */
+void Matrix_MultZero(Vec3f* dest) {
+    MtxF* cmf = sCurrentMatrix;
+
+    dest->x = cmf->xw;
+    dest->y = cmf->yw;
+    dest->z = cmf->zw;
+}
+
+/**
+ * @brief Multiply the vector `(x, 0, 0, 1)` by current.
+ *
+ * I.e. calculate \f[ A(x, 0, 0) + b \f].
+ *
+ * @note Special case of @sa Matrix_MultVec3f with `src = { x, 0, 0 }`; the same assumptions apply.
+ *
+ * @param[in] x multiplier of unit vector in x direction.
+ * @param[out] dest output vector.
+ *
+ * @remark original name: "Matrix_Position_VecX"
+ */
+void Matrix_MultVecX(f32 x, Vec3f* dest) {
+    MtxF* cmf = sCurrentMatrix;
+
+    dest->x = cmf->xw + cmf->xx * x;
+    dest->y = cmf->yw + cmf->yx * x;
+    dest->z = cmf->zw + cmf->zx * x;
+}
+
+/**
+ * @brief Multiply the vector `(0, y, 0, 1)` by current.
+ *
+ * I.e. calculate \f[ A(0, y, 0) + b \f].
+ *
+ * @note Special case of @sa Matrix_MultVec3f with `src = { 0, y, 0 }`; the same assumptions apply.
+ *
+ * @param[in] y multiplier of unit vector in y direction.
+ * @param[out] dest output vector.
+ *
+ * @remark original name is most likely "Matrix_Position_VecY" by analogy with the other two.
+ */
+void Matrix_MultVecY(f32 y, Vec3f* dest) {
+    MtxF* cmf = sCurrentMatrix;
+
+    dest->x = cmf->xw + cmf->xy * y;
+    dest->y = cmf->yw + cmf->yy * y;
+    dest->z = cmf->zw + cmf->zy * y;
+}
+
+/**
+ * @brief Multiply the vector `(0, 0, z, 1)` by current.
+ *
+ * I.e. calculate \f[ A(0, 0, z) + b \f]`.
+ *
+ * @note Special case of @sa Matrix_MultVec3f with `src = { 0, 0, z }`; the same assumptions apply.
+ *
+ * @param[in] z multiplier of unit vector in z direction.
+ * @param[out] dest output vector.
+ *
+ * @remark original name: "Matrix_Position_VecZ"
+ */
+void Matrix_MultVecZ(f32 z, Vec3f* dest) {
+    MtxF* cmf = sCurrentMatrix;
+
+    dest->x = cmf->xw + cmf->xz * z;
+    dest->y = cmf->yw + cmf->yz * z;
+    dest->z = cmf->zw + cmf->zz * z;
+}
+
+/**
+ * @brief Calculates current * (src,1) and writes its x and z components to dest.
+ *
+ * The same as @sa Matrix_MultVec3f, but only applies to the x and z components; the same assumptions apply.
+ *
+ * @note Unlike the previous functions, does *not* just multiply (x, 0, z, 1) and save the x,y,z components.
+ *
+ * @param[in] src input vector.
+ * @param[out] dest output vector.
+ */
+void Matrix_MultVec3fXZ(Vec3f* src, Vec3f* dest) {
+    MtxF* cmf = sCurrentMatrix;
+
+    dest->x = cmf->xw + (cmf->xx * src->x + cmf->xy * src->y + cmf->xz * src->z);
+    dest->z = cmf->zw + (cmf->zx * src->x + cmf->zy * src->y + cmf->zz * src->z);
+}
+
+/**
+ * @brief Copies the matrix src into dest.
+ *
+ * @param[out] dest matrix to copy to.
+ * @param[in] src matrix to copy from.
+ *
+ * @remark original name: "Matrix_copy_MtxF"
+ */
+void Matrix_MtxFCopy(MtxF* dest, MtxF* src) {
+    f32 fv0;
+    f32 fv1;
```

Comment:
probably just `temp1` and `temp2`

Reply:
I think I mainly left them like that in case someone wants to improve the match.

Reply:
I think I'm just going to leave this as-is rather than try and neaten it up any more.